quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/main.bicep

```diff
@@ -0,0 +1,92 @@
+@description('The name of the Azure Application Gateway')
+param AgwName string = 'agw-${uniqueString(resourceGroup().id)}'
+
+@description('The name of the Azure Application Gateway')
+param AkvName string = 'akv-${uniqueString(resourceGroup().id)}'
+
+@description('The name of the certificate to generate for the frontend AGW. Name will be used as a ARecord in a Private DNS zone.')
+param FrontendCertificateName string = 'frontend'
+
+@description('The location to deploy the resources to')
+param Location string = resourceGroup().location
+
+@description('The name of the DNS Private Zone to create')
+param DnsPrivateZoneName string = 'Contoso.local'
+
+resource dns 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: DnsPrivateZoneName
+  location: 'global'
+  
+  resource frontend 'A@2020-06-01' = {
+    name: FrontendCertificateName
+    properties: {
+      ttl: 60
+      aRecords: [
+        {
+          ipv4Address: agw.outputs.agwPip
+        }
+      ]
+    }
+  }
+  
+  resource appdirect 'A@2020-06-01' = {
+    name: 'appdirect'
+    properties: {
+      ttl: 60
+      aRecords: [
+        {
+          ipv4Address: app.outputs.IpAddress
+        }
+      ]
+    }
+  }
+}
+
+var FrontendCertificateFqdn = '${FrontendCertificateName}.${DnsPrivateZoneName}'
+
+resource akv 'Microsoft.KeyVault/vaults@2021-11-01-preview' = {
+  name: AkvName
+  location: Location
+  properties: {
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: subscription().tenantId
+    enableSoftDelete: false
+    enableRbacAuthorization: true
+    accessPolicies: []
+  }
+}
+output akvName string = akv.name
```

Comment:
organize the `outputs` to the end of the file and `vars` between `params` and `resources`